geopandas/geodataframe.py

```diff
@@ -575,6 +576,31 @@ def merge_geometries(block):
 
         return aggregated
 
+    # overrides GeoPandasBase method
+    def explode(self):
+        """
+        Explode multi-part geometries into multiple single geometries.
+
+        Single rows can become multiple rows.
+        This is analogous to PostGIS's ST_Dump(). 
+
+        Returns
+        ------
+        A GeoDataFrame with rows containing MultiPoints expanded into single
+        Point rows. The rest of the original columns are preserved.
+        """
+        exploded_geom = self.geometry.explode().reset_index()
+        exploded_geom.rename(columns={0:'geometry'})
+        df = exploded_geom.merge(
+            self.drop('geometry', axis=1),
```

Comment:
as commented before, we need to use `self._geometry_column_name` here, as it is not required to be 'geomety'